app/event/[id]/page.tsx

```diff
@@ -25,7 +26,9 @@ export async function generateMetadata({ params }: Props): Promise<Metadata> {
   const id = (await params).id;
 
   const queryClient = getQueryClient();
-  const event = await queryClient.fetchQuery(eventOptions(id));
+  const event = await queryClient.fetchQuery(eventOptions(id)).catch((_) => {
+    notFound();
+  });
```

Comment:
avoid mixing async/await will callback style (`.catch`) please

```suggestion
  let event
  try {
    event = await queryClient.fetchQuery(eventOptions(id))
  } catch {
    notFound()
  }
```

Reply:
also you could use an error boundary instead of try/catch

Reply:
Fixed